Cubical/Structures/Poset.agda

```diff
@@ -0,0 +1,189 @@
+{-# OPTIONS --cubical --safe #-}
+
+module Cubical.Structures.Poset where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Logic
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Equiv        hiding   (_■)
+open import Cubical.Foundations.SIP          renaming (SNS-≡ to SNS)
+open import Cubical.Core.Primitives
+open import Cubical.Foundations.HLevels
+open import Cubical.Data.Sigma
+open import Cubical.Data.Sigma.Properties
+
+-- We will adopt the convention of denoting the level of the carrier
+-- set by ℓ₀ and the level of the relation result by ℓ₁.
+private
+  variable
+    ℓ₀ ℓ₁ : Level
+
+Order : (ℓ₁ : Level) → Type ℓ₀ → Type (ℓ-max ℓ₀ (ℓ-suc ℓ₁))
+Order ℓ₁ A = A → A → hProp ℓ₁
+
+order-iso : (M N : Σ (Type ℓ₀) (Order ℓ₁)) →  fst M ≃ fst N → Type (ℓ-max ℓ₀ ℓ₁)
+order-iso (A , _⊑₀_) (B , _⊑₁_) eqv =
+  (x y : A) → [ x ⊑₀ y ⇔ f x ⊑₁ f y ]
+  where
+    f = equivFun eqv
+
+isSetOrder : (ℓ₁ : Level) (A : Type ℓ₀) → isSet (Order ℓ₁ A)
+isSetOrder ℓ₁ A = isSetΠ (λ _ → isSetΠ λ _ → isSetHProp)
+
+Order-is-SNS : SNS {ℓ₀} (Order ℓ₁) order-iso
+Order-is-SNS {ℓ₁ = ℓ₁} {X = X}  _⊑₀_ _⊑₁_ = f , f-equiv
+  where
+    f : order-iso (X , _⊑₀_) (X , _⊑₁_) (idEquiv X) → _⊑₀_ ≡ _⊑₁_
+    f i = funExt λ x → funExt λ y → ⇔toPath (fst (i x y)) (snd (i x y))
+
+    ⇔-prop : isProp ((x y : X) → [ x ⊑₀ y ⇔ x ⊑₁ y ])
+    ⇔-prop = isPropΠ λ x → isPropΠ λ y → snd (x ⊑₀ y ⇔ x ⊑₁ y)
+
+    f-equiv : isEquiv f
+    f-equiv = record { equiv-proof = λ eq → (g eq , sec eq) , h eq }
+      where
+        g : (eq : _⊑₀_ ≡ _⊑₁_)
+          → (x y : X)
+          → ([ x ⊑₀ y ] → [ x ⊑₁ y ]) × ([ x ⊑₁ y ] → [ x ⊑₀ y ])
+        g eq x y = subst (λ { _⊑⋆_ → [ x ⊑⋆ y ] }) eq
+                 , subst (λ { _⊑⋆_ → [ x ⊑⋆ y ] }) (sym eq)
+
+        sec : section f g
+        sec p = isSetOrder _ X _⊑₀_ _⊑₁_ (f (g p)) p
+
+        h : (p : _⊑₀_ ≡ _⊑₁_) → (fib : fiber f p) → (g p , sec p) ≡ fib
+        h p (i , _) = ΣProp≡ A-prop B-prop
+          where
+            A-prop : _
```

Comment:
As you're not giving the type here why not inline the definitions?

Reply:
The motivation for the where clause here is to keep the lines short. I could inline `B-prop` but it's more symmetric this way. Regarding the redundant declarations, I did get rid of them now.

Reply:
Ok, sounds good!